app/src/main/java/com/cornellappdev/scoop/ui/components/post/FirstPage.kt

```diff
@@ -0,0 +1,202 @@
+                    .wrapContentSize()
+            ) {
+                Button(
+                    modifier = Modifier
+                        .size(56.dp),
+                    shape = RoundedCornerShape(30.dp),
+                    enabled = proceedEnabled,
+                    onClick = {
+                        when {
+                            methodOfTransportation.value.isEmpty() -> {
+                                showInvalidMethodMessage = true
+                                proceedEnabled = false
+                                coroutineScope.launch {
+                                    disableMessage()
+                                }
+                            }
+                            departureLocation.value.isEmpty() -> {
+                                showInvalidDepartureMessage = true
+                                proceedEnabled = false
+                                coroutineScope.launch {
+                                    disableMessage()
+                                }
+                            }
+                            arrivalLocation.value.isEmpty() -> {
+                                showInvalidArrivalMessage = true
+                                proceedEnabled = false
+                                coroutineScope.launch {
+                                    disableMessage()
+                                }
+                            }
+                            else -> {
+                                onProceedClicked()
+                            }
+                        }
+                    },
+                    contentPadding = PaddingValues(10.dp),
+                    colors = buttonColors(backgroundColor = Gray)
+                ) {
+                    Icon(
+                        Icons.Outlined.ArrowForward,
+                        contentDescription = stringResource(R.string.arrow_forward_description)
+                    )
+                }
+            }
+        }
+
+        Box {
+            BuildMessage(
+                showInvalidMethodMessage,
+                stringResource(R.string.invalid_method_of_transportation)
+            )
+            BuildMessage(showInvalidDepartureMessage, stringResource(R.string.invalid_departure))
+            BuildMessage(showInvalidArrivalMessage, stringResource(R.string.invalid_arrival))
+        }
+    }
+}
+
+@Composable
+fun TextFields(
+    methodOfTransportation: MutableState<String>,
+    departureLocation: MutableState<String>,
+    arrivalLocation: MutableState<String>,
+    tripState: MutableState<Trip>
+) {
+    Column(
+        modifier = Modifier
+            .background(color = Color.White)
+    ) {
+        Text(
+            text = stringResource(R.string.method_of_transportation),
+            modifier = Modifier
+                .fillMaxWidth(),
+            fontSize = 22.sp
+        )
+        Spinner(methodOfTransportation, listOf("Car", "Uber", "Train", "Bicycle")) {
```

Comment:
Try putting the various methods of transportation into a string array in String.xml so it's easy to change across the app at that one point!